great_expectations/render/renderer_configuration.py

```diff
@@ -490,7 +498,7 @@ def _choose_param_type_for_value(
 
     def add_param(
         self,
-        name: str,
+        name: constr(min_length=1),
```

Comment:
Is this getting dynamically added to a `pydantic` model?
I don't know if it does anything at runtime as a function arg annotation.

Reply:
Yes this adds an attribute to the `params` on `RendererConfiguration`. Yeah I think you are correct that this won't work. `name` already can't be an empty string because it becomes the name of a `BaseModel`. I'm going to leave this as-is.

Reply:
```
great_expectations/render/renderer_configuration.py:501: error: Invalid type
comment or annotation  [valid-type]
            name: constr(min_length=1),
                  ^
great_expectations/render/renderer_configuration.py:501: note: Suggestion: use constr[...] instead of constr(...)
```

type-checker doesn't like this. I'm not sure it really makes sense outside of `pydantic.BaseModel` field annotation.

You *can* enforce method/function calls the same way you do with init args, but it doesn't happen automatically.
https://docs.pydantic.dev/usage/validation_decorator/